core/src/main/resources/defaults/blueprints/cdp-sdx-medium-ha.bp

```diff
@@ -0,0 +1,428 @@
+            "roleType": "KAFKA_BROKER"
+          }
+        ],
+        "serviceConfigs": [
+          {
+            "name": "zookeeper_service",
+            "ref": "zookeeper"
+          },
+          {
+            "name": "offsets.topic.replication.factor",
+            "value": "1"
+          }
+        ],
+        "serviceType": "KAFKA"
+      },
+      {
+        "refName": "ranger",
+        "roleConfigGroups": [
+          {
+            "base": true,
+            "refName": "ranger-RANGER_USERSYNC-BASE",
+            "roleType": "RANGER_USERSYNC"
+          },
+          {
+            "base": true,
+            "refName": "ranger-RANGER_TAGSYNC-BASE",
+            "roleType": "RANGER_TAGSYNC"
+          },
+          {
+            "base": true,
+            "refName": "ranger-RANGER_ADMIN-BASE",
+            "roleType": "RANGER_ADMIN"
+          }
+        ],
+        "serviceConfigs": [
+          {
+            "name": "hdfs_service",
+            "ref": "hdfs"
+          },
+          {
+            "name": "rangeradmin_user_password",
+            "value": "{{{ general.password }}}"
+          },
+          {
+            "name": "rangertagsync_user_password",
+            "value": "{{{ general.password }}}"
+          },
+          {
+            "name": "solr_service",
+            "ref": "solr"
+          },
+          {
+            "name": "rangerusersync_user_password",
+            "value": "{{{ general.password }}}"
+          },
+          {
+            "name": "keyadmin_user_password",
+            "value": "{{{ general.password }}}"
+          }
+        ],
+        "serviceType": "RANGER"
+      },
+      {
+        "refName": "solr",
+        "roleConfigGroups": [
+          {
+            "refName": "solr-SOLR_SERVER-BASE",
+            "roleType": "SOLR_SERVER",
+            "base": true
+          }
+        ],
+        "serviceConfigs": [
+          {
+            "name": "hdfs_service",
+            "ref": "hdfs"
+          },
+          {
+            "name": "zookeeper_service",
+            "ref": "zookeeper"
+          }
+        ],
+        "serviceType": "SOLR"
+      },
+      {
+        "refName": "zookeeper",
+        "serviceType": "ZOOKEEPER",
+        "roleConfigGroups": [
+          {
+            "refName": "zookeeper-SERVER-BASE",
+            "roleType": "SERVER",
+            "base": true
+          }
+        ]
+      },
+      {
+        "serviceType": "KNOX",
+        "refName": "knox",
+        "roleConfigGroups": [
+          {
+            "base": true,
+            "refName": "knox-KNOX_GATEWAY-BASE",
+            "roleType": "KNOX_GATEWAY",
+            "configs": [
+              {
+                "name": "gateway_master_secret",
+                "value": "cloudera1"
+              }
+            ]
+          },
+          {
+            "base": true,
+            "configs": [
+              {
+                "name": "idbroker_master_secret",
```

Comment:
Like above for `gateway_master_secret`, probably should come from Java.

Reply:
created jira for this config to be injected by CB.

Reply:
Implemented in 9450dc701d.